Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -4,56 +4,203 @@
-        let spacer = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.FlexibleSpace, target: nil, action: nil)
-        toolbar.items = [doneItem, spacer]
-
-        let listTableView = UITableView()
-        listTableView.dataSource = self
-        listTableView.delegate = self
-        view.addSubview(listTableView)
-
-        toolbar.snp_makeConstraints { make in
-            let topLayoutGuide = self.topLayoutGuide as! UIView
-            make.top.equalTo(topLayoutGuide.snp_bottom)
-            make.left.right.equalTo(self.view)
-            return
+    var listData = [(item:WKBackForwardListItem, type:BackForwardType)]()
+    
+    init(profile: Profile, backForwardList: WKBackForwardList, isPrivate: Bool) {
+        self.profile = profile
+        self.isPrivate = isPrivate
+        super.init(nibName: nil, bundle: nil)
+        
+        setupTableView()
+        loadSites(backForwardList)
+    }
+    
+    func setupTableView() {
+        tableView.separatorStyle = .None
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.alwaysBounceVertical = false
+        tableView.registerClass(BackForwardTableViewCell.self, forCellReuseIdentifier: BackForwardListCellIdentifier)
+        
+        tableView.backgroundColor = isPrivate ? BackForwardViewUX.BackgroundColorPrivate:BackForwardViewUX.BackgroundColor
+        let blurEffect = UIBlurEffect(style: isPrivate ? .Dark : .ExtraLight)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        tableView.backgroundView = blurEffectView
+    }
+    
+    func loadSites(backForwardList: WKBackForwardList) {
+        let sql = profile.favicons as! SQLiteHistory
+        var urls: [String] = [String]()
+        
+        for page in backForwardList.forwardList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Forward))
         }
-
-        listTableView.snp_makeConstraints { make in
-            make.top.equalTo(toolbar.snp_bottom)
+        
+        var currentRow = 0
+        if let currentPage = backForwardList.currentItem {
+            currentRow = listData.count
+            urls.append(currentPage.URL.absoluteString)
+            listData.append((currentPage, .Current))
+        }
+        for page in backForwardList.backList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Backward))
+        }
+        
+        listData = listData.filter { !(($0.item.title ?? "").isEmpty)}
+        
+        sql.getSitesForURLs(urls).uponQueue(dispatch_get_main_queue()) { result in
+            if let cursor = result.successValue {
+                for cursorSite in cursor {
+                    if let site = cursorSite, let url = site?.url {
+                        self.sites[url] = site
+                    }
+                }
+                self.tableView.reloadData()
+            }
+        }
+        
+        if currentRow > 1 {
```

Comment:
Nit: Instead of having the `if` check here, it would be safer to include this check as a `guard` inside the `scrollTableViewToIndex` method in case we call the method from somewhere else and forget to do the check.